swap-v2-headless-example/index.ts

```diff
@@ -122,32 +123,45 @@ const main = async () => {
   console.log("quoteResponse: ", quote);
 
   // 4. sign permit2.eip712 returned from quote
-  let signature;
+  let signature: Hex | undefined;
```

Comment:
Not sure if this example is only to show the case where permit2 is present. 

If it's a generic example, then the permit2 message signing process should be skipped when `quote.permit2` is `null`

Reply:
ah good catch. added in the logic to skip over this if null